app/controllers/katello/concerns/api/v2/repository_content_controller.rb

```diff
@@ -129,6 +136,19 @@ def filter_by_repos(repos, collection)
         collection.in_repositories(repos)
       end
 
+      def filter_same_by_repos(repos, collection, version_ids, same = true)
+        collection = collection.in_repositories(repos)
+        if same
+          return collection.where(id: collection.select do |resource|
+            resource.in_versions?(version_ids)
+          end)
+        else
+          return collection.where.not(id: collection.select do |resource|
```

Comment:
```suggestion
          collection.where.not(id: collection.select do |resource|
```

Reply:
Also isn't 
```ruby
collection.select do |resource|
     resource.in_versions?(version_ids)
 end
```
fairly expensive ? Will this select run for potentially 5000 rpms ?

Reply:
I think I might have an idea to make this more efficient. Let me try some things out and update the PR here..